tests/integration/abi_roundtrip_test.py

```diff
@@ -103,11 +109,44 @@ def roundtrip_setup(abi_type):
     )
 
 
+def test_abi_types_comprehensive():
+    top_level_names = {
+        tli.split("[")[0] if tli.startswith("pyteal") else tli.split("'")[1]
+        for tli in (
+            str(x) for x in (at[0] if isinstance(at, tuple) else at for at in ABI_TYPES)
+        )
+    }
+
+    def get_subclasses(cls):
+        for subclass in cls.__subclasses__():
+            yield from get_subclasses(subclass)
+            yield subclass
+
+    all_abi_names = {
+        str(at).split("'")[1]
+        for at in (
+            cls
+            for cls in abi.__dict__.values()
+            if isinstance(cls, type)
+            and issubclass(cls, abi.BaseType)
+            and not cls.__abstractmethods__
+            and cls is not abi.Tuple
+        )
+    }
+
+    missing_cases = all_abi_names - top_level_names
+    assert not missing_cases, f"missing round trip tests for {missing_cases}"
+
+
 @pytest.mark.parametrize("abi_type", ABI_TYPES)
 def test_pure_compilation(abi_type):
     print(f"Pure Compilation Test for {abi_type=}")
     abi_type, dynamic_length, roundtripper = roundtrip_setup(abi_type)
 
+    # TODO: Remove once we have necessary bits in SDK for reference types
```

Comment:
I recommend a slightly more communicative approach. Please see the `BAD_TEALS` dict below. I recommend adding a new value variable
```python
GAI_ISSUE_2068 = "https://github.com/algorand/go-algorand-internal/issues/2068"
```
as well as the following keys to `BAD_TEALS`:
```
"account": GAI_ISSUE_2068,
"asset": GAI_ISSUE_2068,
"application": GAI_ISSUE_2068,
```
and this will skip the graviton round-trip tests. It won't skip the pure-TEAL generation though. If that's still broken, we can either see how easy it would be to fix the TEAL generation, or just skip those generations in the same way.